### DISCUSSION TITLE
[Dataimporter] Import to Many-To-One Relation

### DISCUSSION THREAD

**Discussion by Niklasbrandstrom** [pimcore/pimcore#17502] [Q&A / Support]
I need to do an import where i set a lot of relations to both "Many-To-One Relation" and "Many-To-Many" Relation. In the Transformation pipeline i set As Array, and then "Load Dataobject" and see in the transformation result is valid and see my
But now my problem start when setting the Data Target - I can't choose Field Name for the type: Many To Many relation

- Due the lack of documentation of Data Target "Many to many relation" ([Data_Target](https://pimcore.com/docs/platform/Data_Importer/Configuration/Mapping_Configuration/Data_Target/)) my first question is - is it even possible to import to a Many-To-One field.
- And the second question: Please help me understand what i'm doing wrong here, becuase my feeling is that sometimes it works and sometimes its not.

Please help me understand how data target: type many to many works :) (Becuase i have probly a temporary blackout about this)

Thank you in advance

**Comment by fashxp** [comment]
Hi, 
many-to-one relations should be possible using the `direct` data target, 
![image](https://github.com/user-attachments/assets/9e9e2914-8486-4696-b600-9777085cabbc)

Many-to-many relations should also be assignable via the `direct` data target as well as with the `many-to-many` data target. 
The `many-to-many` data target adds two additional options
- also assign data to advanced-many-to-many relation fields
- specify append/merge behavior

Some more you can find here https://github.com/pimcore/data-importer/pull/96. 
Documentation is lacking indeed, maybe someone can provide some additions via a PR ;-).
